### PR TITLE
chore: upgrade semver@7.5.3

### PR DESCRIPTION
**Pull request by ziyad-elabid-nw**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X ] Other, please explain:
At the moment, Eslint uses version ^7.3.5 of semver. The problem arises when launching the command 'npm audit' due to a vulnerability in regular expression denial of service via the function 'new Range'. You can find more information about this vulnerability in the following reference: https://github.com/advisories/GHSA-c2qf-rxjj-qqgw

This pull request (PR) proposes upgrading the semver dependency to version 7.5.3 because it includes the necessary patches for versions greater than or equal to 7.5.2.

#### What changes did you make? (Give an overview)
Update package.json with the appropriate version

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Thanks for the PR!

We're using `semver` as a dev dependency, and with eslint package's versions only so there was no real risk. Either way, an in-range dependency update seems fine to me.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Just waiting for CI to finish before merging.

**Comment by ziyad-elabid-nw** [comment]
> LGTM, thanks!
> 
> Just waiting for CI to finish before merging.

Thanks 💯

**Pull request merged**
